moveit_experimental/jog_arm/src/jog_arm/jog_calcs.cpp

```diff
@@ -72,7 +72,7 @@ JogCalcs::JogCalcs(const JogArmParameters parameters, JogArmShared& shared_varia
 
   resetVelocityFilters();
 
-  jt_state_.name = move_group_.getJointNames();
+  jt_state_.name = joint_model_group_->getJointModelNames();
```

Comment:
It worked on a different robot setup and saw the robot move, but I was kind of hoping that that together with it being a small change meant it would pass the tests here without me having to figure out all the ros pytest stuff, but I guess not so apologies.  It might take a while to debug the test so let me know if I should close and reopen later after getting more time to investigate, since it may take a while.

`joint_model_group_` is assigned on line 62, so I don't think that is the problem.

Reply:
Sorry, I missed this assignment (my IDE wasn't finding usages of the variable at all).
I will have a short look into the test later. Maybe @AndyZe can help as well?

Reply:
I shortly looked into the test: Looks like jog_arm_server crashes after encountering an unknown joint name:
```
[ERROR] ros.moveit_experimental.jog_arm_server: Command joint name unknown.
[ERROR] ros.moveit_core.exceptions: Variable 'virtual_joint' is not known to model 'panda'
Exception thrown.
terminate called after throwing an instance of 'moveit::Exception'
  what():  Variable 'virtual_joint' is not known to model 'panda'
```
As a consequence the test doesn't get feedback and fails with empty `received` buffer.

Reply:
Thanks for testing.  I got a chance to try on the motoman robot, and found that `world` was retrieved as one of the joints, which is likely the cause for failure.  I suppose an easy (but undesirable) solution is to just pass the list of joints as a parameter instead of the move_group.  I'm not sure I know moveit in enough depth to find a good solution but can keep slowly investigating when I have time.

Reply:
Looking into the code, I noticed that `MoveGroupInterface` actually returns variable names when calling getJointNames():
```c++
const std::vector<std::string>& MoveGroupInterface::getJointNames()
{
  return impl_->getJointModelGroup()->getVariableNames();
}
```
Thus, try to use `getVariableNames()`.

Reply:
Thanks, that's probably best then indeed.  The error message I mentioned seems to be there in the master branch as well, so it isn't due to this change.